dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -5069,6 +5018,58 @@ private Contentlet assignIdentifierIfAbsent(final Contentlet contentlet, final C
+
+            // Once the contentlet is saved, it gets refresh from the db. for which the incoming data gets lost.
+            // Therefore here we need to make sure we use the original contentlet that comes with the info passed from the ui.
+            final String hostId = UtilMethods.isSet(contentletRaw.getHost()) ? contentletRaw.getHost() : contentlet.getHost();
+            identifier.setHostId(hostId);
+            if(contentlet.isFileAsset()){
+                try {
+                    if(contentletRaw.getBinary(FileAssetAPI.BINARY_FIELD) == null){
+                        final String binaryIdentifier = contentletRaw.getIdentifier() != null ? contentletRaw.getIdentifier() : StringPool.BLANK;
+                        final String binaryNode = contentletRaw.getInode() != null ? contentletRaw.getInode() : StringPool.BLANK;
+                        throw new FileAssetValidationException("Unable to validate field: " + FileAssetAPI.BINARY_FIELD
+                                + " identifier: " + binaryIdentifier
+                                + " inode: " + binaryNode);
+                    } else {
+                        //We no longer use the old BinaryField to recover the file name.
+                        //From now on we'll recover such value from the field "fileName" presented on the screen.
+                        //The physical file asset is just an internal piece that is mapped to the system asset-name.
+                        //The file per-se no longer can be renamed. We can only modify the asset-name that refers to it.
+                        final String assetName = (String) contentletRaw.getMap()
+                                .get(FileAssetAPI.FILE_NAME_FIELD);
+                        identifier.setAssetName(UtilMethods.isSet(assetName) ?
+                                assetName :
+                                contentletRaw.getBinary(FileAssetAPI.BINARY_FIELD).getName()
+                        );
+                    }
+                } catch (IOException e) {
+                    Logger.error( this.getClass(), "Error handling Binary Field.", e );
+                }
+            } else if ( contentlet.isHTMLPage() ) {
+                //For HTML Pages - The asset name maps to the page URL
+                identifier.setAssetName( htmlPageURL );
+            }
+            if(UtilMethods.isSet(contentletRaw.getFolder()) && !contentletRaw.getFolder().equals(FolderAPI.SYSTEM_FOLDER)){
+                final Folder folder = APILocator.getFolderAPI().find(contentletRaw.getFolder(), systemUser, false);
+                Identifier folderIdent = APILocator.getIdentifierAPI().find(folder.getIdentifier());
+                identifier.setParentPath(folderIdent.getPath());
+            } else {
+                identifier.setParentPath("/");
+            }
+            identifier = APILocator.getIdentifierAPI().save(identifier);
+
+            changedURI = ! oldURI.equals(identifier.getURI());
```

Comment:
intellij gives me a warning about this value not getting modified. Maybe this method a `CheckinContext` Object with the contentlet and other things that are supposed to get  modified

Reply:
Fixed